urlc98c3b97d2_Trimax_venta_tgz-pJ8-com_gesoftware_venta_structures_map_MapTestJ8/src/com/gesoftware/venta/logging/LoggingUtility.java

```diff
@@ -0,0 +1,285 @@
+
+        /* *
+         * METHOD: Logging level enum class constructor
+         * AUTHOR: Eliseev Dmitry
+         * */
+        private LoggingLevel(final int value, final String name) {
+            this.m_Value = value;
+            this.m_Name  = name;
+        } /* End of 'LoggingLevel::LoggingLevel' method */
+
+        /* *
+         * METHOD: Logging level value getter
+         * RETURN: Current logging level value
+         * AUTHOR: Eliseev Dmitry
+         * */
+        public int getValue() {
+            return m_Value;
+        } /* End of 'LoggingLevel::getValue' method */
+
+        /* *
+         * METHOD: Logging level name getter
+         * RETURN: Current logging level value
+         * AUTHOR: Eliseev Dmitry
+         * */
+        public String getName() {
+            return m_Name;
+        } /* End of 'LoggingLevel::getName' method */
+
+        /* *
+         * METHOD: Get logging level by it's name
+         * RETURN: Logging level if success, LEVEL_NONE otherwise
+         *  PARAM: [IN] name - logging level name
+         * AUTHOR: Eliseev Dmitry
+         * */
+        public static LoggingLevel getByName(final String name) {
+            if (name == null)
+                return LoggingLevel.LEVEL_NONE;
+
+            final LoggingLevel loggingLevel = m_LoggingLevels.get(name.toUpperCase());
+            return (loggingLevel == null)?LoggingLevel.LEVEL_NONE:loggingLevel;
+        } /* End of 'LoggingLevel::getByName' method */
+    } /* End of 'LoggingLevel' enum class */
+
+    /* Date formatter */
+    private final SimpleDateFormat m_Formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+    /* Current logging level */
+    private LoggingLevel m_LoggingLevel = LoggingLevel.LEVEL_NONE;
+
+    /* Real logging utility instance */
+    private static final LoggingUtility c_Self = new LoggingUtility();
+
+    /* *
+     * METHOD: Logging utility class constructor
+     * AUTHOR: Eliseev Dmitry
+     * */
+    private LoggingUtility() {
+        m_Formatter.setTimeZone(Calendar.getInstance().getTimeZone());
+        m_Name = "";
+    } /* End of 'LoggingUtility::LoggingUtility' method */
+
+    /* *
+     * METHOD: Sets current logging level
+     *  PARAM: [IN] level - logging level
+     * AUTHOR: Eliseev Dmitry
+     * */
+    private void setLevel(final LoggingLevel level) {
+        m_LoggingLevel = level;
+    } /* End of 'LoggingLevel::setLevel' method */
+
+    /* *
+     * METHOD: Prints message to log file
+     *  PARAM: [IN] level   - message level
+     *  PARAM: [IN] message - message to print
+     * AUTHOR: Eliseev Dmitry
+     * */
+    private void print2File(final String outputString) {
+        /* No output file specified */
+        if (m_OutputFile == null)
+            return;
+
+        try {
+            /* Open file */
+            final FileWriter fileWriter = new FileWriter(m_OutputFile, true);
+            final BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *bufferedWriter*. The resource is referenced at lines 126, 129, 130, 131. The resource is closed at lines 134, 135. There are other execution paths that do not contain closure statements, for example, when BufferedWriter.flush throws an exception. Ensure *bufferedWriter* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).